src/components/ItemFilters/FiltersContainer.tsx

```diff
@@ -1,73 +1,103 @@
 import { ItemFilterData, LocationFilterData } from "../../models/ItemFilterData"
 import ItemFilter from "./ItemFilter"
 import {
-  buildAppliedFiltersString,
-  buildItemFilterQueryString,
-  buildItemsMatchedStringString,
-  parseItemFilterQueryParams,
+  buildAppliedFiltersTagSetData,
+  buildItemFilterQuery,
+  filtersAreApplied,
+  removeValueFromFilters,
 } from "../../utils/itemFilterUtils"
 
 interface ItemFilterContainerProps {
-  itemAggs: Aggregation[]
+  itemAggregations: Aggregation[]
+  handleFiltersChange?: (newAppliedFilterQuery: ItemFilterQueryParams) => void
+  appliedFilters?: AppliedItemFilters
 }
 
-const ItemFilterContainer = ({ itemAggs }: ItemFilterContainerProps) => {
-  const router = useRouter()
+const FiltersContainer = ({
+  itemAggregations,
+  handleFiltersChange,
+  appliedFilters = { location: [], format: [], status: [], year: [] },
+}: ItemFilterContainerProps) => {
   const { isLargerThanLarge, isLargerThanMedium } = useNYPLBreakpoints()
   const filterGroupClassName = isLargerThanLarge
     ? styles.filterGroup
     : styles.filterGroupMobile
 
   const filterData = useRef<ItemFilterData[]>(
-    itemAggs.map((agg: Aggregation) => {
-      if (agg.field === "location") return new LocationFilterData(agg)
-      else return new ItemFilterData(agg)
+    itemAggregations.map((aggregation: Aggregation) => {
+      if (aggregation.field === "location")
+        return new LocationFilterData(aggregation)
+      else return new ItemFilterData(aggregation)
     })
   ).current
 
-  const appliedFilters = useMemo(() => {
-    return parseItemFilterQueryParams(router.query)
-  }, [router.query])
-
-  const appliedFiltersDisplay = buildAppliedFiltersString(
+  const appliedFiltersTagSetData = buildAppliedFiltersTagSetData(
     appliedFilters,
     filterData
   )
+
   const ref = useRef<HTMLDivElement>(null)
+
   useCloseDropDown(() => setWhichFilterIsOpen(""), ref)
 
   const [whichFilterIsOpen, setWhichFilterIsOpen] = useState("")
 
-  const itemsMatched = buildItemsMatchedStringString(router.query)
+  const filtersApplied = filtersAreApplied(appliedFilters)
 
-  const submitFilters = (selection: string[], field: string) => {
-    const newFilters = { ...appliedFilters, [field]: selection }
+  const submitFilters = (selectedFilters: string[], field: string) => {
+    const newFilters = { ...appliedFilters, [field]: selectedFilters }
     const locationFilterData = filterData.find(
       (filter) => filter.field === "location"
     ) as LocationFilterData
-    const url = buildItemFilterQueryString(
+    const itemFilterQuery = buildItemFilterQuery(
       newFilters,
       locationFilterData.recapLocations()
     )
+    handleFiltersChange(itemFilterQuery)
+    setWhichFilterIsOpen("")
+  }
+
+  const clearAllFilters = () => {
+    handleFiltersChange({})
     setWhichFilterIsOpen("")
-    router.push("/search/advanced" + url)
+  }
+
+  const handleRemoveFilterClick = ({ id }: TagSetFilterDataProps) => {
```

Comment:
I mentioned this before but clicking on a tag to remove the filter does not seem to work.

Reply:
I see what happened here. I tried to make the ids uniqe and kebab-cased but forgot that they needed to be the value to allow for proper clearing of the filter. Fixed.